### PR TITLE
Support serializing JWK using RFC 7517

### LINKED ISSUE #16155: JsonWebKey from the Library is not compliant with IETF
**Question:**
Azure.Security.KeyVault.Keys.JsonWebKey returned from the Library as per https://docs.microsoft.com/en-us/dotnet/api/azure.security.keyvault.keys.keyvaultkey.key?view=azure-dotnet does not have kty, kid, key_ops properties for e.g., 

IETF Link: https://tools.ietf.org/html/rfc7517

### PR DESCRIPTION
**Pull request by heaths**
Resolves #16155

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKeyConverter.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+namespace Azure.Security.KeyVault.Keys
+{
+    /// <summary>
+    /// Converts a <see cref="JsonWebKey"/> to or from JSON.
+    /// </summary>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public class JsonWebKeyConverter : JsonConverter<JsonWebKey>
```
Comment (pakrym) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKeyConverter.cs:15]: Can we make this type internal?

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKeyConverter.cs:15]: Can we? I thought they had to be public. Checking...

Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKeyConverter.cs:15]: I see [`GeoJsonConverter`](https://github.com/Azure/azure-sdk-for-net/blob/4a63bdef269573277ad4d7fad3f8aa39a7b50e0f/sdk/core/Azure.Core/src/GeoJson/GeoJsonConverter.cs#L14) is internal, so, sure!

**Comment by heaths** [comment]
@schaabs @christothes @pakrym Bueller?

**Review by christothes — APPROVED** [review]
I started this and then it fell off my radar. -thanks for the reminder!

**Pull request merged**